betterttv.js

```diff
@@ -4098,7 +4078,449 @@ module.exports = function () {
     }
 }
-},{"../helpers/debug":43}],42:[function(require,module,exports){
+},{"../helpers/debug":47}],43:[function(require,module,exports){
+var debug = require("../../debug"),
+    vars = require("../../vars"),
+    tseInnerTemplate = require('../../templates/team_tse-content-inner-div'),
+    followButtonTemplate = require('../../templates/team_follow-button'),
+    loadTeam = require("./team-load-team");
+
+module.exports = function () {
+    if(bttv.settings.get("formatTeamPage") !== true || $("body").attr("data-page") != "teams#show") return;
+    
+    vars.teamCurrentChannel = ($(".js-playing").attr("id")).replace("channel_", "");
+    vars.teamFirstLoad = 1;
+    vars.teamName = window.location.pathname.replace(/\/team\/|\/event\//, "");
+    
+    var teamFollowList = "none",
+        teamFollowedFailedList = "none",
+        teamFollowRequestCount = 0,
+        teamFollowResponseCount = 0,
+        teamFollowInterval = "none";
+        
+    $(window).resize(function() {
+        var vpWidth = $(window).width(),
+            vpHeight = $(window).height(),
+            paddedWidth = vpWidth - 40,
+            paddedHeight = vpHeight - ($("#site_header").outerHeight(true) + 40),
+            teamListHeight = paddedHeight - ($("#team_logo").outerHeight(true) + $("#bttvFollowTeamButton").outerHeight(true) + 2),
+            playerWidth = paddedWidth - ($(".filters.grid.c4").outerWidth(true) + $("#bttvTeamChat").outerWidth(true) + 2),
+            arMultiplier = (playerWidth) / 16,
+            playerHeight = (9 * arMultiplier) + 32;
+            
+        if(paddedWidth < 985) {
+            paddedWidth = 985;
+        }
+
+        $("div.main.discovery.team").css("width", vpWidth+"px");
+        $("div.wrapper.c12.clearfix").css("width", paddedWidth+"px");
+        $("#player_column").css({"width":(playerWidth+2)+"px", "height":paddedHeight+"px"});
+        $("#site_footer").css("width", playerWidth+"px");
+        $("#standard_holder").css({"width":playerWidth+"px", "height":playerHeight+"px"});
+        $("#bttvTeamChat").css("height", (paddedHeight-2)+"px");
+        $("#bttvTeamMemberListOuter").css({"height":teamListHeight+"px"});
+    });
+    
+    var betterTeamInit = function() {
+        var teamCSS = $("<link>", {"href":"//cdn.betterttv.net/style/stylesheets/betterttv-team-page.css?"+bttv.info.versionString(), "id":"betterTwitchTeams", "rel":"stylesheet", "type":"text/css"});
+        
+        if(bttv.settings.get("darkenedMode") === true) {
+            teamCSS = $("<link>", {"href":"//cdn.betterttv.net/style/stylesheets/betterttv-team-page-dark.css?"+bttv.info.versionString(), "id":"betterTwitchTeams", "rel":"stylesheet", "type":"text/css"});
+        }
+        $("body").append(teamCSS);
+        
+        getFeatured();
+        formatTeamPage();
+        //if it wasn't for the bttv.TwitchAPI._ref is null error in Firefox, I would just load the team without the timeout
+        setTimeout(loadTeam, 1000);
```

Comment:
Ok, back to the TwitchAPI._ref is null error hehe... Again, this is specific to Firefox and team pages on Twitch.

While using the published version of BTTV in Firefox, if I visit any team page, I get the "e.TwitchAPI._ref is null" type error. So it's not something that I'm doing per say that's causing the error, all I know is that any function that makes a call to bttv.TwitchAPI.get within the first second or so of a team page loading in Firefox will cause it.

When I test using a local version of BTTV, the error is still present under the same conditions in Firefox, but it becomes "bttv.TwitchAPI._ref is null". The only ways I know to get around it are to either delay the use of bttv.TwitchAPI.get for a second or so, ooor...

If I comment out line 30 from the "_untakeover" function in src/twitch-api (this._ref = null) then everything *appears* to work fine. I am not suggesting that this is a fix (I know it's not), I'm only making the observation because I honestly don't why it's happening. I get the gist of what's going on in twitch-api, but not enough to be able to find the cause or resolve the issue. I was hoping maybe you might have some input @night ?